redaxo/src/core/lib/util/i18n.php

```diff
@@ -412,10 +423,91 @@ private static function loadFile($dir, $locale)
      */
     private static function loadAll($locale)
     {
+        if (self::$cacheLoaded === null) {
+            self::loadCache();
+        }
+
+        if (self::$cacheLoaded && isset(self::$msg[$locale])) {
+            return;
+        }
+
         foreach (self::$directories as $dir) {
             self::loadFile($dir, $locale);
         }
 
         self::$loaded[$locale] = true;
     }
+
+    /**
+     * Loads the cache if not already loaded.
+     */
+    private static function loadCache()
+    {
+        if (rex::getConsole() || rex_backend_login::hasSession() && ($user = rex_backend_login::createUser()) && $user->isAdmin()) {
+            self::$cacheLoaded = false;
+
+            // we need to write the cache file even the cache isn't used by the user
```

Comment:
Warum?

Reply:
damit non-admins den cache direkt verwenden können. Ansonsten würde der nur bei den oben genannten Aktionen geleert werden.
Wenn ich an die DX von Addonentwicklern denke, habe ich sonst das Problem, wenn ich Änderungen vornehme sehe ich die, aber durch den nicht neugeschriebenen Cache der non-Admin nicht.

Dies müsste man nicht machen, wenn es eine Möglichkeit gäbe zu prüfen, ob der Cache neu geschrieben werden müsste.